assets/src/edit-story/app/story/useStoryReducer/reducers/arrangePage.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * Internal dependencies
+ */
+import { isInsideRange, moveArrayElement } from './utils';
+
+/**
+ * Move page in page order with the given id to the given position.
+ *
+ * If new position is outside bounds, nothing happens.
+ * If page is already at the new position, nothing happens.
+ *
+ * Current page remains unchanged (but might have moved in the page order).
+ *
+ * Current selection is unchanged.
+ *
+ * TODO: Handle multi-page re-order when UX and priority is finalized.
+ *
+ * @param {Object} state Current state
+ * @param {Object} payload Action payload
+ * @param {number} payload.pageId Id of page to move to a new position.
+ * @param {number} payload.position Index of where page should be moved to.
+ * @return {Object} New state
+ */
+function arrangePage( state, { pageId, position } ) {
+	const pageIndex = state.pages.findIndex( ( { id } ) => id === pageId );
```

Comment:
Is there's only one Page, should we just return state?

Reply:
That seems like a fair optimization - done in c07622e.